src/rpc/dial.rs

```diff
@@ -143,13 +144,15 @@ impl Service<http::Request<BoxBody>> for ViamChannel {
     fn poll_ready(&mut self, cx: &mut TaskContext<'_>) -> Poll<Result<(), Self::Error>> {
         match self {
             Self::Direct(channel) => channel.poll_ready(cx),
+            Self::DirectPreAuhtorized(channel) => channel.poll_ready(cx),
```

Comment:
I don't love this but it seemed the be the simplest approach in order to get the Result type to work better without adding an Either or having it be a `dyn tonics specific service type with associations that is really annoying to type out`